install-javascript/action.yaml

```diff
@@ -0,0 +1,47 @@
+name: "JavaScript Installer"
+author: "Jaspar Stach <jaspar.stach@greenbone.net>"
+description: "An action to install Greenbone JavaScript Projects"
+inputs:
+  version:
+    description: "Node version that should be installed"
+    default: "16.x"
+  token:
+    description: "Token with GitHub Packages read rights required to install."
+    required: true
+branding:
+  icon: "package"
+  color: "green"
+runs:
+  using: "composite"
+  steps:
+    - run: echo "Checkout Repository"
+      shell: bash
+    - uses: actions/checkout@v3
+    - name: Cache node modules
+      shell: bash
+      id: cache-npm
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-node-modules
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
```

Comment:
I am not sure if this is fine in all cases but imho it's up to the JS devs to work on this.

Reply:
So we should create an input variable for the `key`?

Reply:
I remember that I setting the cache key was not that obvious for JS and yarn with CircleCI in the past. Maybe it is better to use an input variable.